docs/source/developing-volttron/developing-agents/using-concurrency-in-agents.rst

```diff
@@ -0,0 +1,130 @@
+.. _Using-Concurrency-In-Agents:
+
+===================================
+Using Concurrency in Agents
+===================================
+
+When would you want to use concurrency in your agent? You might want to use concurrency if you want to improve performance.
+For example, your agent performs an I/O-bound task such as scraping a website or talking to a slow device (e.g. hard drive or printer)
+and you notice it takes a significant amount to time to complete such a task.
+Or you might have to use concurrency because your agent uses a dependency that has concurrency--which forces your agent
```

Comment:
Threading is another way to fix the IO bound thing, but it is more resource intensive than co routines.